src/seedu/addressbook/Main.java

```diff
@@ -80,15 +81,15 @@ private void exit() {
 
     /** Reads the user command and executes it, until the user issues the exit command.  */
     private void runCommandLoopUntilExitCommand() {
-        Command command;
+        AggregateCommand aggregateCommand;
         do {
             String userCommandText = ui.getUserCommand();
-            command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
+            aggregateCommand = new Parser().parseCommand(userCommandText);
+            CommandResult result = executeCommand(aggregateCommand);
```

Comment:
Can we do this without using polymorphic variables too? I suspect we'll end up with lot of duplicate/boilerplate switch statements but that's the point we are trying to make. e.g.,
```java
switch (Parser.getCommandType(userCommandText)):
ADD:
    AddCommand addCommand = Parser.parseAddCommand(userCommandText);
    ...
DELETE:
   ...
``` 

Perhaps do a dirty PR (i.e., no need for commit organization etc.) to see how the result look like before deciding whether we really want to go that way?

On a related note, the need to make small steps through OOP concepts is less now, due to CS2030. We can decide to use all OOP concepts in level 2, including polymorphism. What do you guys think?

Reply:
> Can we do this without using polymorphic variables too? I suspect we'll end up with lot of duplicate/boilerplate switch statements but that's the point we are trying to make. e.g.,

Does that imply combining the parsing and the execution together? I think the reason we have `AggregateCommand` in the first place was because the commands aren't executed straight away, but is stored in a variable first, hence `AggregateCommand` was created to solve this "storing" issue.

> On a related note, the need to make small steps through OOP concepts is less now, due to CS2030. We can decide to use all OOP concepts in level 2, including polymorphism. What do you guys think?

You are right, I just checked the [web page for cs2030](https://nus-cs2030.github.io/1718-s2/schedule/index.html), they teach that in like week 2 :P.

I think it is OK to expect them to know polymorphism well given the syllabus in cs2030. So I think having a small codebase is more important here, in not making it too overwhelming for students.

Reply:
I agree with Wang Leng, the lecture notes seem to be quite detailed too. The following image is from the CS2030 notes, which explains the need for polymorphism quite well already:

![image](https://user-images.githubusercontent.com/19281514/35656691-a8ad3bd6-0733-11e8-968c-7664a4d87e35.png)

So I think there's perhaps no need for us to make our code to look something like the image above too :P

Reply:
Sure we can leave it be. CS2030 is now a prerequisite for CS2103 and the number of students who took the old route is going to dwindle out in the coming semesters. @eugenepeh sorry I didn't think of this earlier. :-)

Reply:
@damithc oh okay, no problem. So this issue is closed?

Reply:
Yes, please close both issue and PR. May want to point to this discussion when closing the issue.